### DISCUSSION TITLE
GNU compilation error 'hello world'

### DISCUSSION THREAD

**Discussion by LarsKerk** [AMReX-Codes/amrex#3052] [Q&A]
When trying to compile the 'Hello World' example code with GNUmake I get the following error:

sethostname: Use the Network Control Panel Applet to set hostname.
hostname -s is not supported.
sethostname: Use the Network Control Panel Applet to set hostname.
hostname -s is not supported.
process_begin: CreateProcess(NULL, uname, ...) failed.
../../../amrex/Tools/GNUMake/Make.defs:174: pipe: No error

Het systeem kan het opgegeven pad niet vinden.
'grep' is not recognized as an internal or external command,
operable program or batch file.
'grep' is not recognized as an internal or external command,
operable program or batch file.
'printf' is not recognized as an internal or external command,
operable program or batch file.
'tr' is not recognized as an internal or external command,
operable program or batch file.
'tr' is not recognized as an internal or external command,
operable program or batch file.
Loading ../../../amrex/Tools/GNUMake/comps/gnu.mak...
'g++' is not recognized as an internal or external command,
operable program or batch file.
process_begin: CreateProcess(NULL, expr >= 8, ...) failed.
../../../amrex/Tools/GNUMake/comps/gnu.mak:47: pipe: Bad file descriptor
../../../amrex/Tools/GNUMake/comps/gnu.mak:48: *** GCC < 8 not supported.  Stop.

I have no idea why or what goes wrong, any help would be very much appreciated!

**Comment by WeiqunZhang** [comment]
Is this on a Windows machine?

**Comment by LarsKerk** [comment]
Yes, this is on a windows device
________________________________
Van: Weiqun Zhang ***@***.***>
Verzonden: donderdag 1 december 2022 1:46
Aan: AMReX-Codes/amrex ***@***.***>
CC: LarsKerk ***@***.***>; Author ***@***.***>
Onderwerp: Re: [AMReX-Codes/amrex] GNU compilation error 'hello world' (Discussion #3052)


Is this on a Windows machine?

**Reply by WeiqunZhang** [reply]
Could you provide more details?  I am not a Windows user.  So I am just guessing.  Is this Windows Subsystem for Linux?  Do you have gcc and GNU Coreutils installed?

Alternatively, you can also try Visual Studio and CMake.  I have no ideas on how to use these.  But I know some AMReX users are able to do that.  And this might give you some hints.  https://github.com/AMReX-Codes/amrex/blob/development/.github/workflows/windows.yml.

**Reply by LarsKerk** [reply]
The gcc and GNU Coreutils are installed btw, yes

**Comment by LarsKerk** [comment]
The main issue I come across is the fact that the program cannot open the source file AMReX_Config.H, and there being no such file present anywhere in my folders.
________________________________
Van: Weiqun Zhang ***@***.***>
Verzonden: vrijdag 2 december 2022 1:31
Aan: AMReX-Codes/amrex ***@***.***>
CC: LarsKerk ***@***.***>; Author ***@***.***>
Onderwerp: Re: [AMReX-Codes/amrex] GNU compilation error 'hello world' (Discussion #3052)


Could you provide more details? I am not a Windows user. So I am just guessing. Is this Windows Subsystem for Linux? Do you have gcc and GNU Coreutils installed?

Alternatively, you can also try Visual Studio and CMake. I have no ideas on how to use these. But I know some AMReX users are able to do that. And this might give you some hints. https://github.com/AMReX-Codes/amrex/blob/development/.github/workflows/windows.yml.

**Reply by LarsKerk** [reply]
Another error that occurs, is that building 'CMakeLists.txt' creates the error 'No CMAKE_C_COMPILER could be found'
